pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -362,12 +363,14 @@ func (vm *VirtualMachine) getVirtualDeviceByPath(ctx context.Context, diskPath s
 		glog.Errorf("Failed to get the devices for VM: %q. err: %+v", vm.InventoryPath, err)
 		return nil, err
 	}
+
 	// filter vm devices to retrieve device for the given vmdk file identified by disk path
 	for _, device := range vmDevices {
 		if vmDevices.TypeName(device) == "VirtualDisk" {
 			virtualDevice := device.GetVirtualDevice()
 			if backing, ok := virtualDevice.Backing.(*types.VirtualDiskFlatVer2BackingInfo); ok {
-				if backing.FileName == diskPath {
+				if matchVirtualDiskAndVolPath(backing.FileName, diskPath) {
```

Comment:
Nit: How about renaming to virtualDiskMatchesVolPath?

Reply:
Root cause analysis available in issue itself https://github.com/vmware/kubernetes/issues/392